scripts/installer_test.go

```diff
@@ -0,0 +1,340 @@
+	//}, msiBaseArgs...), args...)
+
+	installOut, err := exec.Command("msiexec", args...).CombinedOutput()
+	if !assert.NoError(t, err) {
+		installLogData, err := ioutil.ReadFile(logPath)
+		if assert.NoError(t, err) {
+			t.Logf("MSIExec log:\n============================\n%s", string(installLogData))
+		}
+		t.Logf("MSIExec output:\n============================\n%s", string(installOut))
+		t.Fatal()
+	}
+}
+
+func requireUninstall(t *testing.T, ctx *testcontext.Context) {
+	logPath := ctx.File("log", "uninstall.log")
+	uninstallOut, err := uninstall(logPath)
+	if !assert.NoError(t, err) {
+		uninstallLogData, err := ioutil.ReadFile(logPath)
+		if !assert.NoError(t, err) {
+			t.Logf("MSIExec log:\n============================\n%s\n", string(uninstallLogData))
+		}
+		t.Fatalf("MSIExec output:\n============================\n%s", string(uninstallOut))
+	}
+}
+
+func uninstall(logPath string) ([]byte, error) {
+	if err := stopServices("storagenode", "storagenode-updater"); err != nil {
+		return []byte{}, err
+	}
+
+	args := append([]string{"/uninstall", *msiPathFlag, "/log", logPath}, msiBaseArgs...)
+	return exec.Command("msiexec", args...).CombinedOutput()
+}
+
+func stopServices(names ...string) (err error) {
+	serviceMgr, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+
+	group := new(errgroup.Group)
+	for _, name := range names {
+		name := name
+		group.Go(func() (err error) {
+			service, err := serviceMgr.OpenService(name)
+			if err != nil {
+				return err
+			}
+			defer func() {
+				err = errs.Combine(err, service.Close())
+			}()
+
+			if _, err = service.Control(svc.Stop); err != nil {
+				return err
+			}
+			return waitForStop(service)
+		})
+	}
+
+	return group.Wait()
+}
+
+func waitForStop(service *mgr.Service) error {
+	ctx, _ := context.WithTimeout(context.Background(), 30*time.Second)
+	for {
+		status, err := service.Query()
+		if err != nil {
+			return err
+		}
+
+		if err := ctx.Err(); err != nil {
+			return err
+		}
+
+		switch status.State {
+		case svc.Stopped:
+			return nil
+		default:
+			time.Sleep(500 * time.Millisecond)
+		}
+	}
+}
+
+func readConfigLines(t *testing.T, ctx *testcontext.Context, configPath string) string {
+	configFile, err := os.Open(configPath)
+	require.NoError(t, err)
+	defer ctx.Check(configFile.Close)
+
+	// NB: strip empty lines and comments
+	configBuf := new(bytes.Buffer)
+	scanner := bufio.NewScanner(configFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		line = strings.Trim(line, " \t\n")
+		out := append(scanner.Bytes(), byte('\n'))
+		if len(line) == 0 {
+			continue
+		}
+		if !strings.HasPrefix(line, "#") {
+			_, err := configBuf.Write(out)
+			require.NoError(t, err)
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		require.NoError(t, err)
+	}
+	return configBuf.String()
+}
+
+func releaseUrl(version, name, ext string) string {
+	urlTemplate := "https://github.com/storj/storj/releases/download/{version}/{service}_{os}_{arch}{ext}.zip"
```

Comment:
I'm not sure if we should rely on an external source like github for testing but I'm not sure how to handle this in different way right now.

Reply:
This is an integration test. So I am not negative to using the production download site.